### ISSUE TITLE
Add __version__ to uv python module

### ISSUE THREAD

**Issue by mkleinbort-ic** [astral-sh/uv#2453]
Just discovered this code fails
```python
import uv
uv.__version__
```
With error

`AttributeError: module 'uv' has no attribute '__version__'`

I know it's rare for users to import `uv` as a module, but it was the "obvious" way to check the version when a jupyter notebook is open.

Note that running `uv --version` on the terminal works fine, so does running `!uv --version` on a notebook

**Comment by potiuk** [comment]
IMHO (and this is just comment from a side - I've learned about it when I proposed to use `_version_` in another module), we should not continue this practice and where popular packages do not have them already, we should not add them - because that only increases the confusion about status of `__version__` attribute. It's not only used inconsistently, but also while it's popular in some packages it's status as "standard" is clearly "NO".

It's been proposed in https://peps.python.org/pep-0396/ and **rejected**. 

Instead the standard (and simple and fast) way of retrieving version of the installed package is via importlib-metadata (https://docs.python.org/3/library/importlib.metadata.html#distribution-versions)  and this is the one I think we - as community should promote.

It's a bit hot topic - see https://mail.python.org/archives/list/python-dev@python.org/thread/EY44EMZ2R4RUMUUIVEFFVOXM2H3L427Z/#R4A266556EN26KO7XOZLPQZFPRPH7QXB where the PEP-396 has been rejected - and of course it's up to maintainers to decide, but my personal opinion is - let's stop using it (or at least let's not add it to packages that don't use it already).
